activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -64,12 +64,18 @@ def initialize(constructor = {})
       end
     end
 
-    def default(key = nil)
-      if key.is_a?(Symbol) && include?(key = key.to_s)
-        self[key]
-      else
-        super
-      end
+    def default(*args)
+      key = args.first
+      args[0] = key.to_s if key.is_a?(Symbol)
+      super(*args)
+    end
+
+    def [](key)
+      super(convert_key(key))
+    end
+
+    def fetch(key, *args)
+      super(convert_key(key), *args)
     end
```

Comment:
This duplicates [the existing implementation below](https://github.com/rails/rails/pull/18909/files#diff-9f4b3d7342097a6f22290f147a09b77bR173).